src/RequestHandle.cpp

```diff
@@ -480,4 +538,52 @@ auto curl_write_data(
     return data_length;
 }
 
+auto curl_write_partial_data(
```

Comment:
why not just merge curl_write_data and curl_write_partial_data? i actually can see a nice elegant way where you just keep appending and then IF max download bytes is set you can update the internal member fields and chop/bail out early if you go over

Reply:
Good point! Removed the partial write callback function and integrated the functionality in curl_write_data -- this also meant we didn't need to reset the the callback each time.